app/services/targets/create_service.rb

```diff
@@ -11,13 +11,31 @@ def initialize(current_user, params)
     def call
       raise MaxAllowTargetsError unless targets.count < MAX_ALLOWED_TARGETS
 
-      targets.create!(params)
+      Target.transaction do
```

Comment:
I think it'd be nice if you could adjust specs to properly test this code & make adjustments for the tests to pass. Then this code would be ready for review